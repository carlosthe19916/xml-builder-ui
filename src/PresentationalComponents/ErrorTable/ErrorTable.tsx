import React from "react";
import { Table, TableHeader, TableBody, ICell } from "@patternfly/react-table";
import { ErrorCircleOIcon } from "@patternfly/react-icons";
import {
  Card,
  CardBody,
  Bullseye,
  EmptyState,
  EmptyStateIcon,
  Title,
  EmptyStateBody,
  EmptyStateSecondaryActions,
  Button
} from "@patternfly/react-core";

export interface Props {
  columns: (ICell | string)[];
  retryAction: (event?: any) => void;
}

export interface State {}

class ErrorTable extends React.Component<Props, State> {
  render() {
    return (
      <React.Fragment>
        <Table cells={this.props.columns} rows={[]} aria-label="Error table">
          <TableHeader />
          <TableBody />
        </Table>
        <Card>
          <CardBody>
            <Bullseye>
              <EmptyState>
                <EmptyStateIcon icon={ErrorCircleOIcon} />
                <Title headingLevel="h5" size="lg">
                  ¡Ocurrió un error durante la extracción de datos!.
                </Title>
                <EmptyStateBody>
                  Intente de nuevo o refresque la página.
                </EmptyStateBody>
                <EmptyStateSecondaryActions>
                  <Button
                    variant="link"
                    onClick={this.props.retryAction}
                  >
                    Intentar de nuevo
                  </Button>
                </EmptyStateSecondaryActions>
              </EmptyState>
            </Bullseye>
          </CardBody>
        </Card>
      </React.Fragment>
    );
  }
}

export default ErrorTable;
