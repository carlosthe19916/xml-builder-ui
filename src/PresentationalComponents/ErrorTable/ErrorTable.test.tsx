import React from "react";
import { shallow } from "enzyme";
import ErrorTable from "./ErrorTable";
import { Button } from "@patternfly/react-core";

it("Renders without crashing", () => {
  const retryAction = jest.fn();
  const wrapper = shallow(
    <ErrorTable
      columns={["column1", "column2", "column3"]}
      retryAction={retryAction}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it("Retry action is called when button clicked", () => {
  const retryAction = jest.fn();
  const wrapper = shallow(
    <ErrorTable
      columns={["column1", "column2", "column3"]}
      retryAction={retryAction}
    />
  );

  wrapper.find(Button).simulate("click");
  expect(retryAction.mock.calls.length).toEqual(1);
});
