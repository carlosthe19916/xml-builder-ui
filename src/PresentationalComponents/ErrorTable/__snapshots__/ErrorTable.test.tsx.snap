// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<Fragment>
  <Component
    aria-label="Error table"
    cells={
      Array [
        "column1",
        "column2",
        "column3",
      ]
    }
    rows={Array []}
  >
    <TableHeader />
    <TableBody />
  </Component>
  <Card>
    <CardBody>
      <Bullseye>
        <EmptyState>
          <EmptyStateIcon
            icon={[Function]}
          />
          <Title
            headingLevel="h5"
            size="lg"
          >
            ¡Ocurrió un error durante la extracción de datos!.
          </Title>
          <EmptyStateBody>
            Intente de nuevo o refresque la página.
          </EmptyStateBody>
          <EmptyStateSecondaryActions>
            <Component
              onClick={[MockFunction]}
              variant="link"
            >
              Intentar de nuevo
            </Component>
          </EmptyStateSecondaryActions>
        </EmptyState>
      </Bullseye>
    </CardBody>
  </Card>
</Fragment>
`;
